### DISCUSSION TITLE
Is there a way to slow battery drain?

### DISCUSSION THREAD

**Discussion by cosmicr** [thirtythreeforty/neolink#334] [Q&A] [answered]
I have the following setup:

 - Solar Power
 - Reolink Argus 2
 - Neolink Docker Container
 - Home Assistant Generic Camera

My only issue is that Neolink seems to be continuously streaming the video, even when I'm not viewing it. Or at least it's polling the device, or something, because the battery goes down about 10 to 20% every hour. 

Is there some setting that will have neolink only poll the device when it's requested, or something else to prevent battery drain?

**Comment by QuantumEntangledAndy** [comment, marked as answer]
I think this is currently being tracked in #204. To tldr it this fork always polls. My fork in https://github.com/QuantumEntangledAndy/neolink has some extra features like not polling the stream when not streaming from a client.

However users have said that this is not enough and the battery still drains just by the occasional pinging that goes on between camera and client when not pulling video data from it. 

To progress further we need the camera to talk to us when motion is detected (so that we do not need to stay connected to the camera to know when motion is detected). This might be possible with setting up a mail server and using the cameras mail alerts. But this is not implemented yet.

Also I might be getting a battery camera in the near future so this will help with development of these issues.

**Reply by cosmicr** [reply]
Thanks for the quick reply. I'll watch with close interest 🙂

**Reply by DavidRSullivan** [reply]
The mail server idea isn't bad, but could we implement a mode that doesn't care about motion detection?  It would only connect to the camera on demand.  This would allow us to at minimum support the camera when the user requests to view the stream.  In my case, I use HA to integrate with neolink and while motion detection is great, since battery depletion for me happened in <24 hours with the above QuantunEntangled link, I would use this in on-demand mode right now.

Is that achievable, can I configure it to not poll for motion? Or change the rate of polling myself?

**Comment by QuantumEntangledAndy** [comment]
The `on_client` mode in my fork will eventually do this if you do not also select the `on_motion` pausing. It's not implemented yet though need to find some time for it. There's currently an issue with the way I detect a client is connected or not (it's quite hacky) I would like to find a better way to detect it, and I'd also like to do a full logout when no-one is connected.

I have ordered my own battery camera now so I should be able to test it a bit better and perhaps tune the polling.

**Reply by DavidRSullivan** [reply]
You could implement a REST / style interface to turn on/off the stream, but many people wouldn't easily be able to do this depending on their automation system. 

I do have another issue with my specific integration in that I can't seem to change the port number (the option is there but doesn't seem to work) in the HA Add-On config that installs Neolink into my HA Blue. I believe this is the docker port control.   Currently, I have another non-REO camera linked to that port on my machine, so I have to disable them to get this to work. I will reach out to its author and see if he knows about the issue.
